src/main/java/net/pms/dlna/DLNAMediaInfo.java

```diff
@@ -135,7 +135,7 @@
 	public String aspectRatioVideoTrack;
 	private int videoBitDepth = 8;
 
-	private byte thumb[];
+	private volatile byte[] thumb = null;
```

Comment:
@atamariya No, if everything goes "according to plan" it is only written once. The thumbnail is resized for each request, but that is not stored in that field. Thread safety isn't that simple though. Multiple threads doesn't have to write to an object to create havoc, it's enought that one write and another read at the same time. Even if we assume that a write of a reference is atomic (I have no reason to assume that, it's just to illustrate a point) there still the issue of CPU caches that might be out of sync. As far as I know, what ```volatile``` actually does is force a cache sync among the cores and somehow make the write atomic.

Volatile isn't enought to do it properly here though, and the reason is simple. As long as you have code (and we do here) that checks the state of a given variable before it decides if it should change it (i.e ```thumb == null``` or ```thumb  != null```) volatile isn't enough. One thread could check this and start generating what to write while another does the same check. The first thread then writes its results after the first one has made the check. Because of this, all such code blocks that alter a field/object based on the state of the same field/object must do so inside a synchronized block so that any other thread will wait until the processing is done before it can make its check.